alf/algorithms/ppo_algorithm_test.py

```diff
@@ -39,84 +36,90 @@ def create_algorithm(env, use_rnn=False, learning_rate=1e-1):
             action_spec,
             fc_layer_params=(),
-            continuous_projection_net=StableNormalProjectionNetwork)
+            continuous_projection_net_ctor=StableNormalProjectionNetwork)
         value_net = ValueNetwork(observation_spec, fc_layer_params=())
 
-    optimizer = tf.optimizers.Adam(learning_rate=learning_rate)
+    optimizer = torch.optim.Adam(lr=learning_rate)
+
+    config = TrainerConfig(
+        root_dir="dummy",
+        unroll_length=13,
+        num_updates_per_train_step=4,
+        mini_batch_size=25)
 
     return PPOAlgorithm(
         observation_spec=observation_spec,
         action_spec=action_spec,
+        env=env,
+        config=config,
         actor_network=actor_net,
         value_network=value_net,
-        loss=PPOLoss(
-            action_spec=action_spec, gamma=1.0, debug_summaries=DEBUGGING),
+        loss=PPOLoss(gamma=1.0, debug_summaries=DEBUGGING),
         optimizer=optimizer,
         debug_summaries=DEBUGGING)
 
 
-class PpoTest(tf.test.TestCase):
+class PpoTest(alf.test.TestCase):
     def test_ppo(self):
         env_class = PolicyUnittestEnv
         learning_rate = 1e-1
         iterations = 20
         batch_size = 100
         steps_per_episode = 13
         env = env_class(batch_size, steps_per_episode)
-        env = TFPyEnvironment(env)
 
         eval_env = env_class(batch_size, steps_per_episode)
-        eval_env = TFPyEnvironment(eval_env)
 
         algorithm = create_algorithm(env, learning_rate=learning_rate)
         algorithm.set_summary_settings(summarize_grads_and_vars=DEBUGGING)
-        driver = SyncOffPolicyDriver(env, algorithm)
-        eval_driver = OnPolicyDriver(eval_env, algorithm, training=False)
 
         env.reset()
         eval_env.reset()
-        time_step = driver.get_initial_time_step()
-        policy_state = driver.get_initial_policy_state()
         for i in range(iterations):
-            time_step, policy_state = driver.run(
-                max_num_steps=batch_size * steps_per_episode,
-                time_step=time_step,
-                policy_state=policy_state)
+            algorithm.train_iter()
 
-            algorithm.train(num_updates=4, mini_batch_size=25)
             eval_env.reset()
-            eval_time_step, _ = eval_driver.run(
-                max_num_steps=(steps_per_episode - 1) * batch_size)
+            eval_time_step = unroll(eval_env, algorithm, steps_per_episode - 1)
             logging.info("%d reward=%f", i,
-                         float(tf.reduce_mean(eval_time_step.reward)))
+                         float(eval_time_step.reward.mean()))
 
         eval_env.reset()
-        eval_time_step, _ = eval_driver.run(
-            max_num_steps=(steps_per_episode - 1) * batch_size)
-        logging.info("reward=%f", float(tf.reduce_mean(eval_time_step.reward)))
+        eval_time_step = unroll(eval_env, algorithm, steps_per_episode - 1)
+        logging.info("%d reward=%f", i, float(eval_time_step.reward.mean()))
```

Comment:
Do we need to keep Line 102-104, as they are essentially the same as Line 97-101 for the last iteration?